parallel-rdp/video_interface.hpp

```diff
@@ -72,6 +72,15 @@ struct ScanoutOptions
 		bool divot_filter = true;
```

Comment:
The code is generally well written and organized, however, there are a few improvements that could be made:

1. **Commenting**: There are some parts of the code that lack comments, making it difficult to understand what certain blocks of code are doing. Adding comments would greatly enhance the readability and maintainability of the code. For example, the `decode_vi_registers` function could use some comments explaining what it does and how it works.

2. **Naming Conventions**: The naming convention for variables and functions is inconsistent. Some variables use snake_case while others use camelCase. It's recommended to stick to one naming convention for consistency and readability. 

3. **Error Handling**: The code lacks error handling. It's recommended to add checks for potential errors and handle them appropriately. For example, in the `set_vi_register` function, there should be a check to ensure that the provided `reg` value is valid.

4. **Code Duplication**: There is some code duplication in the code. For example, the `vram_fetch_stage` and `aa_fetch_stage` functions have similar code. It's recommended to create a helper function to avoid code duplication.

5. **Magic Numbers**: There are some magic numbers in the code. It's recommended to replace these magic numbers with named constants to improve readability and maintainability.

6. **Include Necessary Headers**: Ensure all necessary headers are included. For example, if you're using `std::string`, you should include `<string>`. 

7. **Use of `auto`**: Consider using `auto` to simplify code and make it more readable, especially when the type of the variable is obvious from the context.

8. **Use of `enum class`**: Consider using `enum class` instead of `enum` for stronger type safety and better scope management.